isisd/isis_nb_config.c

```diff
@@ -1956,6 +1960,39 @@ int lib_interface_isis_area_tag_modify(struct nb_cb_modify_args *args)
 	return NB_OK;
 }
 
+/*
+ * XPath: /frr-interface:lib/interface/frr-isisd:isis/vrf
+ */
+int lib_interface_isis_vrf_modify(struct nb_cb_modify_args *args)
+{
+	struct interface *ifp;
+	struct vrf *vrf;
+	const char *ifname, *vrfname, *vrf_name;
+	struct isis *isis = NULL;
+
+	if (args->event == NB_EV_VALIDATE) {
+		/* libyang doesn't like relative paths across module boundaries
+		 */
+		ifname = yang_dnode_get_string(args->dnode->parent->parent,
+					       "./name");
+		vrfname = yang_dnode_get_string(args->dnode->parent->parent,
+						"./vrf");
+		vrf = vrf_lookup_by_name(vrfname);
+		assert(vrf);
+		ifp = if_lookup_by_name(ifname, vrf->vrf_id);
+
+		if (!ifp)
+			return NB_OK;
+
+		vrf_name = yang_dnode_get_string(args->dnode, NULL);
+		isis = isis_lookup_by_vrfname(vrf_name);
+		if (isis == NULL)
+			return NB_ERR_VALIDATION;
```

Comment:
Please write something to `args->errmsg` so that the user knows why the validation failed.

Reply:
Error message is added for it..